config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java

```diff
@@ -78,6 +80,20 @@ private void printRulePhases(List<WindupRuleProvider> allProviders)
         LOG.info("Rule Phases: [\n" + rulePhaseSB.toString() + "]");
     }
 
+    private void checkForDuplicates(List<WindupRuleProvider> providers)
+    {
+        // using a map so that we can easily pull out the previous value later (in the case of a duplicate)
+        Set<WindupRuleProvider> dupeSet = new HashSet<>(providers.size());
+        for (WindupRuleProvider provider : providers)
+        {
+            if (dupeSet.contains(provider))
+            {
+                throw new WindupException("Found two providers with the same id: " + provider.getID());
```

Comment:
Yeah, possibly. Although I expect that most of the time the class will be WindupRuleProviderBuilder.